[JSC] rename Array#group/Array#groupByToMap to Object.groupBy/Map.groupBy

dcrousso opened the pull request:
#### 8f5c359bbaeb0eb87946cea9ccdb8c52a2c1f143
<pre>
[JSC] rename Array#group/Array#groupByToMap to Object.groupBy/Map.groupBy
<a href="https://bugs.webkit.org/show_bug.cgi?id=258680">https://bugs.webkit.org/show_bug.cgi?id=258680</a>

Reviewed by Yusuke Suzuki.

Match the latest changes to the &quot;array grouping&quot; spec[1] (proposal[2]).

1. <a href="https://tc39.es/proposal-array-grouping/">https://tc39.es/proposal-array-grouping/</a>
2. <a href="https://github.com/tc39/proposal-array-grouping/">https://github.com/tc39/proposal-array-grouping/</a>

* Source/JavaScriptCore/builtins/ArrayPrototype.js:
(group): Deleted.
(groupToMap): Deleted.
* Source/JavaScriptCore/builtins/MapConstructor.js: Added.
(groupBy):
* Source/JavaScriptCore/builtins/ObjectConstructor.js:
(groupBy): Added.
* Source/JavaScriptCore/runtime/ArrayPrototype.cpp:
(JSC::ArrayPrototype::finishCreation):
* Source/JavaScriptCore/runtime/MapConstructor.cpp:
(JSC::MapConstructor::finishCreation):
* Source/JavaScriptCore/runtime/ObjectConstructor.cpp:
(JSC::ObjectConstructor::finishCreation):
* Source/JavaScriptCore/runtime/OptionsList.h:

* JSTests/stress/array-group-by-null-or-undefined.js: Removed.
* JSTests/stress/map-groupBy.js: Renamed from array-groupBy.js.
* JSTests/stress/object-groupBy.js: Renamed from array-groupByToMap.js.

* Source/JavaScriptCore/CMakeLists.txt:
* Source/JavaScriptCore/DerivedSources.make:
* Source/JavaScriptCore/DerivedSources-input.xcfilelist:
* Source/JavaScriptCore/DerivedSources-output.xcfilelist:
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/265632@main">https://commits.webkit.org/265632@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e06e89ed505b3eba84a085b620b71f1c0a9359e6

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/11443 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/11647 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/12000 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/13088 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/10918 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/14032 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/11629 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/13800 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/11606 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/12476 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/9693 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/13508 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/9768 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/10382 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/17546 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/9733 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/10839 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/10536 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/13735 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/10863 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/10942 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/9012 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/11534 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/10117 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/3107 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2752 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/14390 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/11862 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/10798 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/2880 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

Constellation reviewed (approved):
r=me, and can you update test262 expectations after this change?

dcrousso committed:
[JSC] rename Array#group/Array#groupByToMap to Object.groupBy/Map.groupBy
https://bugs.webkit.org/show_bug.cgi?id=258680

Reviewed by Yusuke Suzuki.

Match the latest changes to the "array grouping" spec[1] (proposal[2]).

1. https://tc39.es/proposal-array-grouping/
2. https://github.com/tc39/proposal-array-grouping/

* Source/JavaScriptCore/builtins/ArrayPrototype.js:
(group): Deleted.
(groupToMap): Deleted.
* Source/JavaScriptCore/builtins/MapConstructor.js: Added.
(groupBy):
* Source/JavaScriptCore/builtins/ObjectConstructor.js:
(groupBy): Added.
* Source/JavaScriptCore/runtime/ArrayPrototype.cpp:
(JSC::ArrayPrototype::finishCreation):
* Source/JavaScriptCore/runtime/MapConstructor.cpp:
(JSC::MapConstructor::finishCreation):
* Source/JavaScriptCore/runtime/ObjectConstructor.cpp:
(JSC::ObjectConstructor::finishCreation):
* Source/JavaScriptCore/runtime/OptionsList.h:

* JSTests/stress/array-group-by-null-or-undefined.js: Removed.
* JSTests/stress/map-groupBy.js: Renamed from array-groupBy.js.
* JSTests/stress/object-groupBy.js: Renamed from array-groupByToMap.js.

* Source/JavaScriptCore/CMakeLists.txt:
* Source/JavaScriptCore/DerivedSources.make:
* Source/JavaScriptCore/DerivedSources-input.xcfilelist:
* Source/JavaScriptCore/DerivedSources-output.xcfilelist:
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/265632@main
